travis/docker.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+if [ "${DOCKER_IMAGE_NAME}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_IMAGE_NAME has not been provided"
+  exit 128
+fi
+
+if [ "${DOCKER_USERNAME}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_USERNAME has not been provided"
+  exit 128
+fi
+
+if [ "${DOCKER_PASSWORD}" = "" ]
+then
+  echo "Docker deployment error. Env DOCKER_PASSWORD has not been provided"
+  exit 128
+fi
+
+echo "Login into docker..."
+echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
+
+CURRENT_VERSION=$(npm run version --silent)
+
+echo "Creating image ${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}"
+docker build -t "${DOCKER_IMAGE_NAME}:${CURRENT_VERSION}" .
+
+if [ "$TRAVIS_BRANCH" = "$MAIN_BRANCH" ]
+then
+    echo "Building for ${$MAIN_BRANCH} branch..."
+    echo "Docker build latest version"
+    docker tag ${DOCKER_IMAGE_NAME}:${CURRENT_VERSION} ${DOCKER_IMAGE_NAME}:latest
+    docker push "${DOCKER_IMAGE_NAME}:latest"
```

Comment:
`latest` may be confusing as it's an alpha, not an official release. People may use that instead of the `release` one. 

I would have 2 tags for alpha 

- ${CURRENT_VERSION}-alpha 
- "${CURRENT_VERSION}-alpha-$(date +%Y%m%d%H%M)"

The second one is for people that want a specific alpha. If there is a new alpha with the same version but you have the old one locally, Docker doesn't download it (without removing the local image). If the client specifies the latest timestamp it will be downloaded.

Reply:
great, i look make more sense 👍